sled-agent/src/sled_agent.rs

```diff
@@ -352,41 +353,44 @@ impl SledAgent {
             )
             .await?;
 
-        // Initialize early networking
-        info!(log, "Attempting to load early network config from bootstore");
-        // Pull the early network config out of the bootstore
-        let boundary_switch_addrs = if let Some(serialized_config) =
-            bootstore.get_network_config().await?
-        {
-            let mut early_networking = EarlyNetworkSetup::new(&log);
-            info!(
-                log,
-                concat!(
-                    "Loaded serialized early network config from bootstore ",
-                    "with generation {}"
-                ),
-                serialized_config.generation,
-            );
+        // Get our rack network config from the bootstore; we cannot proceed
+        // until we have this, as we need to know which switches have uplinks to
+        // correctly set up services.
+        let get_network_config = || async {
+            let serialized_config = bootstore
+                .get_network_config()
+                .await
+                .map_err(|err| BackoffError::transient(err.to_string()))?
+                .ok_or_else(|| {
+                    BackoffError::transient(
+                        "Missing early network config in bootstore".to_string(),
+                    )
+                })?;
 
-            let early_network_config: EarlyNetworkConfig = serialized_config
-                .try_into()
-                .map_err(|e| Error::EarlyNetworkDeserialize(e))?;
-            info!(
-                log,
-                "Successfully deserialized network config: {:?}",
-                early_network_config
-            );
+            let early_network_config =
+                EarlyNetworkConfig::try_from(serialized_config)
```

Comment:
This is interesting. It will perpetually retry until it can successfully deserialize. That is useful property when the code has been updated, but the new data has not been replicated yet. It would also have the effect of possibly never succeeding, but I guess that's fine also as the error message should be indicative of the problem.